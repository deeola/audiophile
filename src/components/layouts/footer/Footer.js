import React from "react";
import FooterNav from "./FooterNav";
import facebook from "../../../assets/shared/desktop/icon-facebook.svg";
import twitter from "../../../assets/shared/desktop/icon-twitter.svg";
import instagram from "../../../assets/shared/desktop/icon-instagram.svg";

function Footer(props) {
  return (
    <div className="footer">
      <FooterNav />
      <div className="footerSecondDiv">
        <p className="footerSecondText">
          Audiophile is an all in one stop to fulfill your audio needs. We're a
          small team of music lovers and sound specialists who are devoted to
          helping you get the most out of personal audio. Come and visit our
          demo facility - weâ€™re open 7 days a week.
        </p>
      </div>
      <div className="copyIcon">
        <p className="copyRight">Copyright 2021. </p>
        <div className="icons">
          <div className="iconDiv">
            <img src={facebook} alt="facebookicon"></img>
          </div>
          <div className="iconDiv">
            <img src={twitter} alt="twittericon"></img>
          </div>
          <div className="iconDiv">
            <img src={instagram} alt="instagramicon"></img>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Footer;
