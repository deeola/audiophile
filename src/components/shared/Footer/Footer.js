import React from "react";
import LogoComponent from "../Navbar/LogoComponent";
import {
  List,
  NavList,
  Copyright,
  Icon,
  SocialMediaIcons,
  SubText,
  FooterContainer,
  FooterNavContainer,
  FooterDescription,
  FooterControl,
  FooterList
} from "./footerstyling";
import { library } from "@fortawesome/fontawesome-svg-core";
import logo from "../../../assets/shared/desktop/logo.svg";

import {
  faFacebook,
  faInstagram,
  faTwitter,
} from "@fortawesome/free-brands-svg-icons";
import Links from "../Navbar/Links";
library.add(faFacebook, faTwitter, faInstagram);

const Footer = () => {
  const copy = new Date().getFullYear();

  return (
    <FooterContainer role={'footer'}>
      <FooterControl>
        <FooterNavContainer>
          <FooterList to={"/"}>
            <LogoComponent img={logo} />
          </FooterList>
          <NavList>
            <List  to="/" >Home</List>
            <List to="/Headphones" >Headphones</List>
            <List to="/Speakers" >Speakers</List>
            <List to="/Earphones"  >Earphones</List>
          </NavList>
        </FooterNavContainer>
        <SubText data-testid='subheading'>
          Audiophile is an all in one stop to fulfill your audio needs. We're a
          small team of music lovers and sound specialists who are devoted to
          helping you get the most out of personal audio. Come and visit our
          demo facility - weâ€™re open 7 days a week.
        </SubText>
        <FooterDescription>
          <Copyright data-testid='year'>
            Copyright {copy}. All Rights Reserved
          </Copyright>
          <SocialMediaIcons>
            <Icon icon={faFacebook} />
            <Icon icon={faTwitter} />
            <Icon icon={faInstagram} />
          </SocialMediaIcons>
        </FooterDescription>
      </FooterControl>
    </FooterContainer>
  );
};

export default Footer;
