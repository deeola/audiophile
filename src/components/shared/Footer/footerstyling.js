import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Link } from "react-router-dom";

const orange = "#d87d4a";
const verylightgray = "#ffffff";

export const FooterContainer = styled.footer`
  background: black;
  padding: 70px 0px;
`;

export const FooterControl = styled.div`
  width: 80%;
  margin: auto;

  display: flex;
  flex-direction: column;

  @media (max-width: 490px) {
    align-items: center;
}

`;

export const FooterNavContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  @media (max-width: 802px) {

    flex-direction: column;
    align-items: unset;
}

@media (max-width: 490px) {
    align-items: center;
}



`;

export const FooterList = styled(Link)`
text-transform: uppercase;
  list-style: none;
  text-decoration: none;
  font-weight: 700;
  font-size: 13px;
  color: white;
  @media (max-width: 802px) {

    width: 30%;
}

@media (max-width: 490px) {

    width: 100%;
}
  
  
`

export const FooterDescription = styled.div`
  display: flex;
  justify-content: space-between;
@media (max-width: 802px) {
   margin-top: 50px;
}
@media (max-width: 490px) {
    flex-direction: column;
    text-align:center;
   
 }
`;

export const SubText = styled.p`
  font-style: normal;
  font-weight: 500;
  font-size: 15px;
  line-height: 25px;
  /* or 167% */

  color: #ffffff;

  mix-blend-mode: normal;
  opacity: 0.5;
  width: 50%;
  margin: 40px 0px 0px 0px; 


  @media (max-width: 802px) {
    width: 100%;
}

@media (max-width: 490px) {
    text-align:center;
 }

`;

export const SocialMediaIcons = styled.div`
  
`;

export const Icon = styled(FontAwesomeIcon)`
  color: white;
  font-size: 20px;
  margin-right: 10px;
  cursor: pointer;

  &:hover {
    color: ${orange};
  }

  @media (max-width: 490px) {
   
    margin-right: 20px;
 }
`;

export const Copyright = styled.p`
  font-style: normal;
  font-weight: 700;
  font-size: 15px;
  line-height: 25px;
  /* identical to box height, or 167% */

  color: #ffffff;

  mix-blend-mode: normal;
  opacity: 0.5;

  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  padding-top: 40px;

  span {
    color: ${orange};
  }

  @media (max-width: 802px) {
      align-items: unset;
      padding-top: 0px;
}

@media (max-width: 490px) {
    align-items: center;
    text-align:center;
    padding-top: 0px;
    margin-bottom: 40px;
 }
`;

export const NavList = styled.ul`
  color: ${verylightgray};
  width: 40%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;

  @media (max-width: 802px) {

    width: 70%;
    margin: 30px 0px 0px 0px;
    align-items: left;
    padding-left: 0px;
}

@media (max-width: 490px) {
   flex-direction: column;
}
`;
export const List = styled(Link)`
  text-transform: uppercase;
  list-style: none;
  text-decoration: none;
  font-weight: 700;
  font-size: 13px;
  color: white;
  &:hover {
    color: ${orange};
    cursor: pointer;
  }

  @media (max-width: 802px) {

    margin-left:0px;
}

@media (max-width: 490px) {
    margin-top: 20px;
    font-weight: 700;
font-size: 13px;
line-height: 25px;
/* identical to box height, or 192% */

letter-spacing: 2px;
text-transform: uppercase;
 }

 
`;


