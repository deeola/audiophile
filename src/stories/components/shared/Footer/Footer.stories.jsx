import React from "react";
import { BrowserRouter } from "react-router-dom";
import Footer from "../../../../components/shared/Footer/Footer";


export default {
  title: "Components/FOOTER",
  component:Footer,
 
};

const Template = (args) => (
  <BrowserRouter>
    <Footer {...args} />
  </BrowserRouter>
);

export const Primary = Template.bind({});
Primary.args = {
  primary: true,
};