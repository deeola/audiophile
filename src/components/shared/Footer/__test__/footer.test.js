import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import reactDom from "react-dom";
import { BrowserRouter } from "react-router-dom";
import Footer from "../Footer";

const MockFooter = () => {
  return (
    <BrowserRouter>
      <Footer />
    </BrowserRouter>
  );
};

it("renders footer component", () => {
  const footer = document.createElement("footer");

  reactDom.render(<MockFooter />, footer);
});

//get footer role

describe("tests for texts in the footer", () => {
  it("gets the footer role", () => {
    const { getByRole } = render(<MockFooter />);
    expect(getByRole("footer")).toBeInTheDocument();
  });

  it("checks if subheading is in document", () => {
    const { getByTestId } = render(<MockFooter />);
    expect(getByTestId("subheading")).toBeInTheDocument();
  });

  it("checks for copyright year", () => {
    const { getByTestId } = render(<MockFooter />);
    expect(getByTestId("year")).toHaveTextContent(/2022/i);
  });
});


//creates snapshot

describe('creates snapshot', () => {
    it('all footersnapshot', () => {
        const tree = renderer.create(<MockFooter />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})