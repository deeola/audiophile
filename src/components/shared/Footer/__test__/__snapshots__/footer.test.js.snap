// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates snapshot all footersnapshot 1`] = `
<footer
  className="sc-kDDrLX hawjIB"
  role="footer"
>
  <div
    className="sc-iqcoie jwZRhm"
  >
    <div
      className="sc-crXcEl fROixB"
    >
      <a
        className="sc-evZas hsvjUp"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="sc-hKMtZM"
          data-testid="logo"
          src="logo.svg"
        />
      </a>
      <ul
        className="sc-bjUoiL gYA-dba"
      >
        <a
          className="sc-idiyUo ddvggH"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="sc-idiyUo ddvggH"
          href="/Headphones"
          onClick={[Function]}
        >
          Headphones
        </a>
        <a
          className="sc-idiyUo ddvggH"
          href="/Speakers"
          onClick={[Function]}
        >
          Speakers
        </a>
        <a
          className="sc-idiyUo ddvggH"
          href="/Earphones"
          onClick={[Function]}
        >
          Earphones
        </a>
      </ul>
    </div>
    <p
      className="sc-ksZaOG bKAMYG"
      data-testid="subheading"
    >
      Audiophile is an all in one stop to fulfill your audio needs. We're a small team of music lovers and sound specialists who are devoted to helping you get the most out of personal audio. Come and visit our demo facility - weâ€™re open 7 days a week.
    </p>
    <div
      className="sc-breuTD hlBqBn"
    >
      <p
        className="sc-fEOsli dSVPdz"
        data-testid="year"
      >
        Copyright 
        2022
        . All Rights Reserved
      </p>
      <div
        className="sc-hAZoDl crfYFd"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-facebook sc-fnykZs jYfIbL"
          data-icon="facebook"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.8 90.69 226.4 209.3 245V327.7h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.3 482.4 504 379.8 504 256z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-twitter sc-fnykZs jYfIbL"
          data-icon="twitter"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M459.4 151.7c.325 4.548 .325 9.097 .325 13.65 0 138.7-105.6 298.6-298.6 298.6-59.45 0-114.7-17.22-161.1-47.11 8.447 .974 16.57 1.299 25.34 1.299 49.06 0 94.21-16.57 130.3-44.83-46.13-.975-84.79-31.19-98.11-72.77 6.498 .974 12.99 1.624 19.82 1.624 9.421 0 18.84-1.3 27.61-3.573-48.08-9.747-84.14-51.98-84.14-102.1v-1.299c13.97 7.797 30.21 12.67 47.43 13.32-28.26-18.84-46.78-51.01-46.78-87.39 0-19.49 5.197-37.36 14.29-52.95 51.65 63.67 129.3 105.3 216.4 109.8-1.624-7.797-2.599-15.92-2.599-24.04 0-57.83 46.78-104.9 104.9-104.9 30.21 0 57.5 12.67 76.67 33.14 23.72-4.548 46.46-13.32 66.6-25.34-7.798 24.37-24.37 44.83-46.13 57.83 21.12-2.273 41.58-8.122 60.43-16.24-14.29 20.79-32.16 39.31-52.63 54.25z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-instagram sc-fnykZs jYfIbL"
          data-icon="instagram"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</footer>
`;
